### PR TITLE
[RLlib]: Cleanup examples folder: Add example restoring 1 of n agents from a checkpoint.

### PR DESCRIPTION
**Pull request by simonsays1980**
## Why are these changes needed?

Restoring certain agents from checkpoint is a frequent use case and we should provide examples for this scenario. This PR is adding such an example in the new API. stack. The example does the following:

1. Training of `n` agents on `Pendulum-v1` `MultiEnv`.
2. Choosing the best checkpoint by return. 
3. Loading the module state for `policy 0` from this checkpoint.
4. Training the agents with `policy 0` restored from checkpoint. 

This example shows that training further on from a restored checkpoint - even for only a single agent - results in faster convergence.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/BUILD [hunk 1]
```diff
@@ -2732,6 +2732,15 @@ py_test(
 #    args = ["--enable-new-api-stack", "--num-agents=2", "--as-test", "--framework=torch", "--stop-reward=-100.0", "--num-cpus=4"],
 # )
 
+py_test(
```
Comment (sven1977) [inline rllib/BUILD:2735]: Ah, good point. I was thinking we should add this under `examples/checkpoints`.
What do you think?

File: rllib/examples/multi_agent/restore_1_of_n_agents_from_checkpoint.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+    # Add a simple multi-agent setup.
+    if args.num_agents > 0:
+        base_config.multi_agent(
+            policies={f"p{i}" for i in range(args.num_agents)},
+            policy_mapping_fn=lambda aid, *a, **kw: f"p{aid}",
+        )
+
+    # Augment the base config with further settings and train the agents.
+    results = run_rllib_example_script_experiment(base_config, args)
```
Comment (sven1977) [inline rllib/examples/multi_agent/restore_1_of_n_agents_from_checkpoint.py:112]: Awesome! Proves that this utility can be used NOT only at the end of a script, but also in the middle to provide new information/data for subsequent calls!

File: rllib/examples/multi_agent/restore_1_of_n_agents_from_checkpoint.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
+"""Simple example of loading module weights for 1 of n agents from checkpoint.
```
Comment (sven1977) [inline rllib/examples/multi_agent/restore_1_of_n_agents_from_checkpoint.py:1]: nit: Let's not use "simple".

"An example script showing how to load RLModule weights for 1 out of n agents from a checkpoint" ?

File: rllib/examples/multi_agent/restore_1_of_n_agents_from_checkpoint.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
```
Comment (sven1977) [inline rllib/examples/multi_agent/restore_1_of_n_agents_from_checkpoint.py:2]: Can we add a tiny paragraph here saying the usual:

```
This example:
- runs a multi-agent Pendulum experiment with ... policies ... blabla
- saves a checkpoint of the used MultiAgentRLModule every blabla iterations
- stops the experiment after the agents reach a combined return of ...
- picks the best of both trained policies (based on episode return) and restores only the corresponding RLModule.
- runs a second experiment with the restored RLModule (single-agent) .... blabla
```

**Review by sven1977 — APPROVED** [review]
Looks good to me! Just a few nits on comments/docstrings.

Awesome example! One more down. :)

File: rllib/examples/multi_agent/restore_1_of_n_agents_from_checkpoint.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
++---------------------+------------+----------------------+--------+
+| Trial name          | status     | loc                  |   iter |
+|---------------------+------------+----------------------+--------+
+| PPO_env_7c6be_00000 | TERMINATED | 192.168.1.111:101257 |     26 |
++---------------------+------------+----------------------+--------+
+
++------------------+-------+-------------------+-------------+-------------+
+|   total time (s) |    ts |   combined return |   return p0 |   return p1 |
++------------------+-------+-------------------+-------------+-------------|
+|          86.7995 | 13312 |          -395.822 |    -315.359 |    -325.237 |
```
[the remark this answers is not in the document]
Comment (simonsays1980) [reply rllib/examples/multi_agent/restore_1_of_n_agents_from_checkpoint.py:43]: I don't know, but I remember I ran into this with the off-policy multi-agent settings last week. So, needs a clean-up for sure.

[the remark this answers is not in the document]
Comment (simonsays1980) [reply rllib/examples/multi_agent/restore_1_of_n_agents_from_checkpoint.py:43]: I will remove this for now until this is fixed: better no output as this weird one.

File: rllib/examples/multi_agent/restore_1_of_n_agents_from_checkpoint.py (hunk 3 above)
Comment (simonsays1980) [reply rllib/examples/multi_agent/restore_1_of_n_agents_from_checkpoint.py:1]: Yup, not simple for everyone lol. I know what you mean, its actually quite some complexity to make this possible in MA scenarios - and so powerful.

**Pull request merged**